src/amber/exceptions/exceptions.cr

```diff
@@ -65,8 +65,12 @@ module Amber
 
     module Validator
       class ValidationFailed < Base
-        def initialize(errors)
-          super("Validation failed. #{errors}")
+        def initialize(@errors : Array(Amber::Validators::Error))
+          super("Validation failed. #{@errors}")
+        end
+
+        def errors
```

Comment:
This could be [a getter instead](https://crystal-lang.org/api/0.27.2/Object.html#getter%28%2Anames%29-macro)